gamestonk_terminal/portfolio/portfolio_model.py

```diff
@@ -333,19 +342,97 @@ def get_rolling_beta(
+    bdte = bcash / (df["holdings"][0] - bcash)
+    edte = ecash / (df["holdings"][-1] - ecash)
+    if sum(d_debt) > 0:
+        t_debt = (
+            f"Beginning debt to equity was {bdte:.2%} and ending debt to equity was"
+            f" {edte:.2%}. Debt adds risk to a portfolio by amplifying the gains and losses when"
+            " equities change in value."
+        )
+        if bdte > 1 or edte > 1:
+            t_debt += " Debt to equity ratios above one represent a significant amount of risk."
+    else:
+        t_debt = "Debt was not used this year. This reduces this risk of the portfolio."
+    string = (
+        f"Your portfolio's performance for the period was {df['return'][-1]:.2%}. This was"
+        f" {'greater' if df['return'][-1] > df[('Market', 'Return')][-1] else 'less'} than"
+        f" the market return of {df[('Market', 'Return')][-1]:.2%}. The variance for the"
+        f" portfolio is {np.var(df['return']):.2%}, while the variance for the market was"
+        f" {np.var(df[('Market', 'Return')]):.2%}. Since our portfolio's variance was"
+        f" {v_com} than the market's variance, our returns should be {v_com} than those"
+        f" of the market. {t_debt} The following report details various analytics from the"
+        f" portfolio. Read below to see the moving beta for a stock."
+    )
+    return string
+
+
+def get_beta_text(df: pd.DataFrame) -> str:
+    """Get beta summary for a dataframe
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        The beta history of the stock
+
+    Returns
+    ----------
+    t : str
+        The beta history for a ticker
+    """
+    betas = df[list(filter(lambda score: "beta" in score, list(df.columns)))]
+    high = betas.idxmax(axis=1)
+    low = betas.idxmin(axis=1)
+    string = (
+        "Beta is how strongly a portfolio's movements correlate with the market's movements."
+        " A stock with a high beta is considered to be riskier, with the average being one."
+        f" The beginning beta for the period was {portfolio_helper.beta_word(df['total'][0])}"
+        f" at {df['total'][0]:.2f}. This went"
+        f" {'up' if df['total'][-1] > df['total'][0] else 'down'} to"
+        f" {portfolio_helper.beta_word(df['total'][-1])} at {df['total'][-1]:.2f} by the end"
+        f" of the period. The ending beta was pulled {'up' if df['total'][-1] > 1 else 'down'} by"
+        f" {portfolio_helper.clean_name(high[-1] if df['total'][-1] > 1 else low[-1])}, which had"
+        f" an ending beta of {df[high[-1]][-1] if df['total'][-1] > 1 else df[low[-1]][-1]:.2f}."
+        f" To {'lower' if df['total'][-1] > 1 else 'raise'} the beta of the portfolio and"
+        f" {'decrease portfolio risk' if df['total'][-1] > 1 else 'increase potential return'}"
+        f" invest more heavily in"
+        f" {portfolio_helper.clean_name(low[-1] if df['total'][-1] > 1 else high[-1])} which had"
```

Comment:
Im not a fan of this.  You are basically just breaking down a single factor (also average beta is not 1 -- that just means your movements are exactly correlated to the market.  If you really wanted, you could include the individual stock betas on the graph or just some kind of table.  

This also kind of reads as investment advice.

Reply:
I do believe the average of all market beta is 1: https://en.wikipedia.org/wiki/Beta_(finance). Also my goal is to expand this and break down lots of different factors in detail. My hope is that this is a report that new investors can use to learn more about finance. 
Also, I do see your point about investment advice. I think to avoid this we tell them the facts and don't tell them what to do. Good catch!

Reply:
> I do believe the average of all market beta is 1: https://en.wikipedia.org/wiki/Beta_(finance). Also my goal is to expand this and break down lots of different factors in detail. My hope is that this is a report that new investors can use to learn more about finance.

Value weighted beta (by definition) is 1.  Thats not the same as average beta.  If average beta was 1, everyone would beat the market.

Reply:
Maybe we can let the more in depth report be added with a flag.